common/views/components/Header/DesktopSignIn.tsx

```diff
@@ -0,0 +1,111 @@
+import DropdownButton from '../DropdownButton/DropdownButton';
+import { BorderlessLink } from '../BorderlessClickable/BorderlessClickable';
+import AlignFont from '../styled/AlignFont';
+import { user as userIcon } from '../../../icons';
+
+const LinkList = styled.div`
+  a {
+    display: block;
+    text-decoration: none;
+    margin-bottom: 16px;
+    &:hover {
+      text-decoration: underline;
+    }
+    &:last-child {
+      margin-bottom: 0;
+    }
+  }
+`;
+
+const DesktopSignIn: FC = () => {
+  const { state, user } = useUser();
+
+  return state === 'initial' ? null : (
+    <>
+      {state === 'signedout' && (
+        <>
+          <span className="display-none headerLarge-display-block">
+            <BorderlessLink
+              iconLeft={userIcon}
+              text={
+                <span
+                  className={classNames({
+                    [font('hnr', 6)]: true,
+                  })}
+                >
+                  Library sign in
+                </span>
+              }
+              href="/account"
+            />
+          </span>
+          <span className="display-none headerMedium-display-block headerLarge-display-none">
+            <DropdownButton
+              label=""
+              iconLeft={userIcon}
+              id="signedin-dropdown"
+              buttonType="borderless"
+            >
+              <span
+                className={classNames({
+                  [font('hnr', 6)]: true,
+                })}
+              >
+                <AlignFont>
+                  <a
+                    href="/account"
+                    onClick={event => {
+                      // This is a very hacked together piece of work that allows us to read this cookie
+                      // and respond to it in the identity app
+                      event.preventDefault();
+                      document.cookie = `returnTo=${window.location.pathname}; path=/`;
+                      window.location.href = event.currentTarget.href;
+                    }}
+                  >
+                    Sign in to your library account
+                  </a>
+                </AlignFont>
+              </span>
+            </DropdownButton>
+          </span>
+        </>
+      )}
+      {state === 'signedin' && user && (
+        <DropdownButton
+          label={
+            <span
+              className={classNames({
+                [font('hnr', 6)]: true,
+              })}
+            >
+              {user.firstName.charAt(0).toLocaleUpperCase()}
+              {user.lastName.charAt(0).toLocaleUpperCase()}
+            </span>
+          }
+          iconLeft={userIcon}
+          id="signedin-dropdown"
+          buttonType="borderless"
+        >
+          <span
+            className={classNames({
+              [font('hnr', 6)]: true,
+            })}
+          >
+            <AlignFont>
+              <LinkList>
+                <a href="/account">Library account</a>
```

Comment:
I'd probably make this into a styled-component that extends `Space` in order to get the margin bottom, instead of adding it [on the LinkList anchors](https://github.com/wellcomecollection/wellcomecollection.org/pull/7024/files#diff-b6add31ad5e6b1ca8f8e6218d771e5420c75941e94c1c6f4694c01c6185036c6R14). Or just wrapping it in a Space.

Reply:
I've moved it into a [styled component](https://github.com/wellcomecollection/wellcomecollection.org/pull/7024/files#diff-b6add31ad5e6b1ca8f8e6218d771e5420c75941e94c1c6f4694c01c6185036c6R13) and used that